best_practices/security.rst

```diff
@@ -173,6 +182,25 @@ the same ``getAuthorEmail`` logic you used above:
 
             return false;
         }
+
+        /**
+         * Checks if the user token has the given role taking into account the
+         * entire role hierarchy defined by the application.
+         */
+        private function userHasRole(TokenInterface $userToken, $roleName)
+        {
+            if (null === $this->roleHierarchy) {
+                return false;
```

Comment:
this is wrong. You should still use ``$userToken->getRoles()`` (which are the only reachable roles in this case).